src/JSInterop/Microsoft.JSInterop/src/Infrastructure/DotNetDispatcher.cs

```diff
@@ -292,6 +294,32 @@ public static void EndInvokeJS(JSRuntime jsRuntime, string arguments)
             }
         }
 
+        /// <summary>
+        /// Accepts the byte array data being transferred from JS to DotNet.
+        /// </summary>
+        /// <param name="jsRuntime">The <see cref="JSRuntime"/>.</param>
+        /// <param name="id">Identifier for the byte array being transfered.</param>
+        /// <param name="data">Byte array to be transfered from JS.</param>
+        public static void SupplyByteArray(JSRuntime jsRuntime, long id, byte[] data)
+        {
+            if (id == 0)
+            {
+                // Starting a new transfer, clear out previously stored byte arrays
+                jsRuntime.ResetByteArraysToBeRevived();
+            }
+            else if (id != (jsRuntime.ByteArraysToBeRevived.Count + 1))
+            {
+                throw new ArgumentOutOfRangeException($"Element id '${id}' cannot be added to the byte arrays to be revived with length '${jsRuntime.ByteArraysToBeRevived.Count}'.");
+            }
+            else if (data.Length + jsRuntime.ByteArraysToBeRevivedByteLength > (31*1024)) // TODO; get this limit from SignalR somehow?
```

Comment:
```suggestion
            else if (data.Length - 32k >  jsRuntime.ByteArraysToBeRevivedByteLength) // TODO; get this limit from SignalR somehow?
```

To avoid int overflows.

Reply:
I believe this should be:

```suggestion
            else if (32k - data.Length <  jsRuntime.ByteArraysToBeRevivedByteLength) // TODO; get this limit from SignalR somehow?
```